src/query/interpreter.cpp

```diff
@@ -432,6 +434,93 @@ class CoordQueryHandler final : public query::CoordinatorQueryHandler {
     }
   }
 
+  void DemoteInstanceToReplica(std::string_view instance_name) override {
+    auto status = coordinator_handler_.DemoteInstanceToReplica(instance_name);
+    switch (status) {
+      using enum memgraph::coordination::DemoteInstanceCoordinatorStatus;
+      case NO_INSTANCE_WITH_NAME:
+        throw QueryRuntimeException("No instance with such name!");
+      case NOT_COORDINATOR:
+        throw QueryRuntimeException("DEMOTE INSTANCE query can only be run on a coordinator!");
+      case NOT_LEADER: {
+        auto const maybe_leader_coordinator = coordinator_handler_.GetLeaderCoordinatorData();
+        auto const *common_message = "Couldn't demote instance to replica since coordinator is not a leader!";
+        if (maybe_leader_coordinator) {
+          throw QueryRuntimeException("{} Current leader is coordinator with id {} with bolt socket address {}",
+                                      common_message, maybe_leader_coordinator->coordinator_id,
+                                      maybe_leader_coordinator->bolt_server.SocketAddress());
+        }
+        throw QueryRuntimeException(
+            "{} Try contacting other coordinators as there might be leader election happening or other coordinators "
+            "are down.",
+            common_message);
+      }
+      case RAFT_LOG_ERROR:
+        throw QueryRuntimeException(
+            "Couldn't demote instance to replica since raft server couldn't append the log. "
+            "Coordinator may not be leader anymore!");
+      case RPC_FAILED:
+        throw QueryRuntimeException(
+            "Couldn't demote instance to replica because current main instance couldn't unregister replica!");
+      case LOCK_OPENED:
+        throw QueryRuntimeException(
+            "Couldn't successfully finish action demote instance to replica because the last action didn't finish "
+            "successfully!");
+      case FAILED_TO_OPEN_LOCK:
+        throw QueryRuntimeException("Couldn't demote instance to replica as cluster didn't accept start of action!");
+      case FAILED_TO_CLOSE_LOCK:
+        throw QueryRuntimeException(
+            "Couldn't demote  instance to replica as cluster didn't accept successful finish of action!");
+      case SUCCESS:
+        break;
+    }
+  }
+
+  void ForceResetClusterState() override {
+    auto status = coordinator_handler_.ForceResetClusterState();
+    switch (status) {
+      using enum memgraph::coordination::ForceResetClusterStateStatus;
+      case NOT_COORDINATOR:
+        throw QueryRuntimeException("Force reset cluster state query can only be run on a coordinator!");
+      case SHUTTING_DOWN:
+        throw QueryRuntimeException("Force reset cluster aborted as coordinator is shutting down!");
+      case ACTION_FAILED:
+        throw QueryRuntimeException("Check logs for more details, one action didn't complete successfully!");
```

Comment:
Which action? Improve message

Reply:
There are two different actions that can fail, for example sending SwapUUID, DemoteToReplica. I wanted to put them under same bin, but I will see if I can split it and introduce different error message for those parts

Reply:
resolved on call not to include message